packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -388,6 +392,9 @@ export class Combobox
   /** Specifies the size of the component. */
   @property({ reflect: true }) scale: Scale = "m";
 
+  /** When `true`, provides a toggle for selecting all items. Does not apply to `selection-mode single`. */
+  @property({ reflect: true }) selectAllEnabled = false;
```

Comment:
Brainstorm: I kinda wonder if instead of creating a new property to enable this, we just add a new `selection-mode` called `all` which implies `multiple` and also adds the ability to select all items?  Thoughts on that?  @jcfranco @geospatialem @SkyeSeitz @ashetland

Reply:
I undid the removal of `select-all-enabled` in favor of having the `Select All` checkbox as a default for now. I'll wait for @jcfranco to chime in after the feedback left by @ashetland of this possibly being a disruption to existing behavior (and needing a way to disable that).